src/main/java/InfrastructureManager/Master.java

```diff
@@ -1,37 +1,138 @@
+                new Thread(runner,"MainRunner").start();
+                break;
+            }
+        }
+    }
+
+    /**
+     * Method that given a certain Scenario, runs the corresponding ScenarioRunner and adds
+     * it to the list of running Runners. If called on a running ScenarioRunner, it will restart
+     * it
+     * @param scenario Scenario to be run
+     */
+    public void runScenario(Scenario scenario) {
+        ScenarioRunner scenarioRunner = getRunner(scenario);
+        if (running.contains(scenarioRunner)) { //If running then stop it and re-run
+            scenarioRunner.exit();
+            running.remove(scenarioRunner);
+        }
+        scenarioRunner.setScenario(scenario);
+        running.add(scenarioRunner);
+        new Thread(scenarioRunner).start(); // Run the scenario in another thread
+
+    }
+
+    /**
+     * Method to pause a running ScenarioRunner, given the scenario that is running
+     * @param scenario Running scenario to be paused
+     */
+    public void pauseScenario(Scenario scenario) {
+        ScenarioRunner scenarioRunner = getRunner(scenario);
+        if (running.contains(scenarioRunner)) {
+            scenarioRunner.pause();
         }
     }
 
+    /**
+     * Method to resume a paused ScenarioRunner, given its scenario
+     * @param scenario Paused scenario to be resumed
+     */
+    public void resumeScenario(Scenario scenario) {
+        ScenarioRunner scenarioRunner = getRunner(scenario);
+        if (running.contains(scenarioRunner)) {
+            scenarioRunner.resume();
+        }
+    }
+
+    /**
+     * Method for finding a ScenarioRunner in the Runner list, based on the scenario is
+     * supposed to run according to the configuration file
+     * @param scenario Scenario for which to find the ScenarioRunner
+     * @return ScenarioRunner that will run the given scenario
+     * @throws IllegalArgumentException If there is no scenarioRunner configured to run the
+     * given scenario
+     */
+    private ScenarioRunner getRunner(Scenario scenario) {
+        ScenarioRunner scenarioRunner;
+        for (Runner runner : runnerList) {
+            try {
+                scenarioRunner = (ScenarioRunner) runner;
+                if (scenarioRunner.getScenarioName().equalsIgnoreCase(scenario.getName())){
+                    return scenarioRunner;
+                }
+            } catch (ClassCastException e) {
+                continue;
+            }
+        }
+        throw new IllegalArgumentException("There is no runner configured for the given scenario");
+    }
+
+    /**
+     * Singleton method for getting the only instance of the class
+     * @return The instance of the Master Class
+     */
+    public static Master getInstance() {
+        if (instance == null) {
+            instance = new Master();
+        }
+        return instance;
+    }
+
+    public static void main(String[] args) {
+        Master.getInstance().startMainRunner();
```

Comment:
I think the main should wait for threads to finish (join).

Reply:
I thought this was the simplest way to achieve the join, but I am not 100% sure is the best. I didnt wanna expose or save the other threads, given that the exit command is made for closing all of them